### PR TITLE
Billing dashboard

### LINKED ISSUE #2394: Implement billing dashboard page
* Contact info
* Credit cards
* Recent billing activity
* Account balance

### PR DESCRIPTION
**Pull request by na3d**
Closes #2394 

image

### CODE REVIEW

File: src/billing/layouts/DashboardPage.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import { setSource } from '~/actions/source';
+import { account, invoices } from '~/api';
+
+
+export class DashboardPage extends Component {
+  static async preload({ dispatch, getState }) {
+    if (!Object.keys(getState().api.account).length) {
+      await dispatch(account.one());
+    }
+    await dispatch(invoices.all());
```
Comment (eatonphil) [inline src/billing/layouts/DashboardPage.js]: I'd collect these into one await:

```
static async preload({ dispatch, getState }) {
  const requests = [invoices.all()];
  if (!Object.keys(getState().api.account).length) {
    requests.push(account.one());
  }

  await Promise.all(requests.map(r => dispatch(r)));
```

File: src/billing/layouts/DashboardPage.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  }
+
+  async componentDidMount() {
+    const { dispatch } = this.props;
+    await dispatch(setSource(__filename));
```
Comment (eatonphil) [inline src/billing/layouts/DashboardPage.js]: No need for this to be async and no need to await the dispatch.

**Comment by eatonphil** [comment]
Tests?

File: src/billing/layouts/DashboardPage.js (hunk 2 above)
Comment (na3d) [reply src/billing/layouts/DashboardPage.js]: Funny how the linter caught the other one, but not that one.

**Pull request closed without merging**